package.json

```diff
@@ -27,6 +29,11 @@
   "dependencies": {
     "axios": "^1.4.0",
     "csv-reader": "^1.0.12",
-    "ts-command-line-args": "^2.5.0"
+    "dotenv": "^16.0.3",
+    "jwt-decode": "^3.1.2",
+    "openid-client": "^5.4.2",
+    "ts-command-line-args": "^2.5.0",
+    "winston": "^3.8.2",
```

Comment:
We're using pino in the other PDC project -- any particular reason not to normalize on that?

Reply:
There was a [brief discussion on Zulip](https://chat.opentechstrategies.com/#narrow/stream/75-PDC/topic/Logging.20in.20node/near/165086), but broadly, `pino-pretty` is an inconvenient extra step - especially around argument passing - and Winston offers first class pretty formatting support and human readable timestamps. I wouldn't want to use locale time strings in prod, but for dev-facing-only scripts, I find them to be more readable.

Reply:
Ahh got it I missed it -- I do believe  pino pretty / timestamp formats can be enabled as part of the config [pretty config](https://github.com/pinojs/pino/blob/HEAD/docs/pretty.md).

Timestamp is just using a [customPrettifier](https://github.com/pinojs/pino-pretty#options) that would convert to whatever timestamp you want.

Not asking you to rewrite using pino or anything, but wanna just be sure it's clear that pino I believe is pretty much feature complete here and there is *some* cost in adding a new logging package that diverges from the project's existing one, since now the team has to become familiar with both.

For context I had initially used winston on the service repo long long ago and we chose pino after some discussion.

Reply:
Aha, I didn't realize it could be done in-app! That's a strong argument.

> Not asking you to rewrite using pino or anything

Hopefully that wouldn't be too big a lift; you're welcome to take a pass on that while I'm working on wrapping up Candid / testing https://github.com/PhilanthropyDataCommons/service/pull/373 !

> For context I had initially used winston on the service repo long long ago and we chose pino after some discussion.

Oh, interesting! Do you have a link to that discussion? What were the tradeoffs?

Reply:
I LIED!  I didn't do the initial logging at all and was just merging OTS projects in my head.  But anyway here's the original issue / related commit logs https://github.com/PhilanthropyDataCommons/service/issues/41

Reply:
I'm OK with the timestamp being RFC-ized by default. The reason for not doing it out of the box is it breaks pino's performance claims. I think having a single JSON line is a feature whereby `grep` works to find my exact log statements of interest. If it's broken into multiple lines by default, now I have add `-C 100` or whatever. Regardless, there may be newlines within a single message, so this point may be moot.

Reply:
Right, that format makes sense in prod! I don't think it fits well in this repo, though.

All that said, at some point I think we're going to want a more sophisticated logging platform than syslog and grep. We don't need to have that conversation now, though.

Reply:
I rewrote the Winston logging commit to use pino instead, and split it out into its own PR: #48.